Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/function/AzureFunctionMvpModel.java

```diff
@@ -131,28 +126,29 @@ public void stopFunction(String sid, String appId) {
     /**
      * List all the Function Apps in selected subscriptions.
      *
-     * @param forceReload flag indicating whether force to fetch latest data from server
+     * @param force flag indicating whether force to fetch latest data from server
      * @return list of Function App
      */
     @AzureOperation(
         name = "function.list.subscription|selected",
         type = AzureOperation.Type.SERVICE
     )
-    public List<ResourceEx<FunctionApp>> listAllFunctions(final boolean forceReload) {
-        final List<ResourceEx<FunctionApp>> functions = new ArrayList<>();
-        List<Subscription> subs = AzureMvpModel.getInstance().getSelectedSubscriptions();
-        if (subs.size() == 0) {
-            return functions;
-        }
-        Observable.from(subs).flatMap((sd) ->
-                Observable.create((subscriber) -> {
-                    List<ResourceEx<FunctionApp>> functionList = listFunctionsInSubscription(sd.subscriptionId(), forceReload);
-                    synchronized (functions) {
-                        functions.addAll(functionList);
-                    }
-                    subscriber.onCompleted();
-                }).subscribeOn(Schedulers.io()), subs.size()).subscribeOn(Schedulers.io()).toBlocking().subscribe();
-        return functions;
+    public List<ResourceEx<FunctionApp>> listAllFunctions(final boolean... force) {
+        return AzureMvpModel.getInstance().getSelectedSubscriptions().parallelStream()
+            .flatMap((sd) -> listFunctionsInSubscription(sd.subscriptionId(), force).stream())
+            .collect(Collectors.toList());
+    }
+
+    @NotNull
+    @AzureOperation(
+        name = "function.list.java|subscription|selected",
+        type = AzureOperation.Type.SERVICE
+    )
+    @Preload
```

Comment:
List Azure resources is only valid when user is signed in. however, users may not signed in preload time.(Even if user signed in before and restart IntelliJ, our auth manager may not be initialized). We may need to add condition for preload event and add more pre load triggers.

Reply:
> Even if user signed in before and restart IntelliJ, our auth manager may not be initialized
no, auth manager will read from file? anyway, it can work normally.

> We may need to add condition for preload event and add more pre load triggers.
as far as I know for this moment, we only need to re-trigger preloading when user sign in after startup/changed subscriptions.
we can complete to cover more cases step by step in the future.